compiler/src/iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -778,6 +778,69 @@ struct FuseWidenOperands : public OpRewritePattern<Op> {
   }
 };
 
+struct DotToMul : public OpRewritePattern<mhlo::DotOp> {
+  using OpRewritePattern<mhlo::DotOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mhlo::DotOp op,
+                                PatternRewriter &rewriter) const override {
+    auto lhs = op.lhs();
+    auto rhs = op.rhs();
+    auto lhsTy = lhs.getType().cast<RankedTensorType>();
```

Comment:
This cast may assert/not required for op AFAIK. Dyn_cast and notifyMatchFailure?

Reply:
Added.